### ISSUE TITLE
StorageManager does not respect `path` prop

### ISSUE THREAD

**Issue by johnpc** [aws-amplify/amplify-ui#5008]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Storage (Storage Manager)

### How is your app built?

Next.js

### What browsers are you seeing the problem on?

Safari

### Please describe your bug.

I am using the `StorageManager` to upload files to S3. Specifically I am using the `path` prop, which according to [the docs](https://ui.docs.amplify.aws/react/connected-components/storage/storagemanager) is

```
A path to put files in the s3 bucket. This will be prepended to the key sent to s3 for each file.
```

However, when I upload files, I am not observing the file uploaded under the specified `path`.

### What's the expected behaviour?

Because `src/app/page.tsx` specifies `path={"folder"}`

```typescript
<StorageManager
  acceptedFileTypes={["image/*"]}
  path={"folder"}
  ...
/>
```

I expect the file uploaded to be uploaded to the S3 bucket nested under a folder named `folder`.

However, the file is uploaded to S3, but not nested under any path. The file is uploaded to the root of the s3 bucket.

### Help us reproduce the bug!

I've created a repro repo so you can see the bug in action: https://github.com/johnpc/feb13-storagemanager

Run the following bash commands to set up the environment

```bash
git clone https://github.com/johnpc/feb13-storagemanager
cd feb13-storagemanager
npm install
npx amplify sandbox
npm run dev
```

Then navigate to http://localhost:3000 and upload a file via the file manager. You'll then find your file at the root of your S3 bucket (not at the specified path)

### Code Snippet

```typescript
// Put your code below this line.

```

**Comment by owenleej** [comment]
I am having this problem but today is the first time I have ever tried using the StorageManager. Is this bug present if I temporarily install an older version?

**Comment by johnpc** [comment]
You can downgrade to v5 `npm i aws-amplify@^5 @aws-amplify/ui-react-storage@^2` but that's a different major version and might affect other parts of your app.

**Comment by calebpollman** [comment]
@johnpc @owenleej Fix has been released in `@aws-amplify/ui-react-storage@3.0.14`
